File: src/de/hsh/dennis/model/actors/Player.java
package de.hsh.dennis.model.actors;

import common.actor.Actor;
import common.actor.Direction;
import javafx.scene.image.Image;

public class Player extends Actor {

    //GameLogic
    private int health = 100;
    private int score = 0;

    //Visuals
    private Image skin_current;
    private final double posX = Config.Player.posX;
    private final double posY = Config.Player.posY;


    public Player() {
        super("player_standard.png", "de/hsh/dennis/resources/actors/Player/");
        setSkinToDefault();
    }

    public void changeSkin(Direction dir) {
        switch (dir) {
            case Left:
                skin_current = Config.Player.skin_left;
                break;
            case Right:
                skin_current = Config.Player.skin_right;
                break;
            case Up:
                skin_current = Config.Player.skin_up;
                break;
            case Down:
                skin_current = Config.Player.skin_down;
                break;

        }
    }

    // --- Utils ----------------------------------------------------------------------------------

    public void damage(int d) {
        if ((getHealth() - d) <= 0) {
            setHealth(0);
        } else {
            setHealth(getHealth() - d);
        }
    }

    public void heal(int h) {
        if ((getHealth() + h) >= 100) {
            setHealth(100);
        } else {
            setHealth(getHealth() + h);
        }
    }


    // --- Getter & Setter ------------------------------------------------------------------------

    public void setSkinToDefault() {
        skin_current = Config.Player.skin_standard;
    }

    public Image getSkin_current() {
        return skin_current;
    }

    public double getPosX() {
        return posX;
    }

    public double getPosY() {
        return posY;
    }

    public int getHealth() {
        return health;
    }

    public void setHealth(int health) {
        this.health = health;
    }

    public int getScore() {
        return score;
    }

    public void setScore(int score) {
        this.score = score;
    }
}

File: Games/Dennis/src/de/hsh/dennis/model/Spawn/actors/Player.kt
package de.hsh.dennis.model.Spawn.actors

import common.actor.Actor
import common.actor.Direction
import de.hsh.dennis.model.Spawn.SkinConfig

class Player : Actor(SkinConfig.Player.skin_standard_path,
                     SkinConfig.Player.posX,
                     SkinConfig.Player.posY) {
    // --- Getter & Setter ------------------------------------------------------------------------
    var direction = Direction.Non
    private var lastLR = Direction.Non

    fun changeSkin(dir: Direction) {
        fun setSkin(playerSkin: String, direction: Direction, lastLR: Direction = direction) {
            setCurrentImage(playerSkin)
            this.direction = direction
            if (lastLR != Direction.Non) {
                this.lastLR = lastLR
            }
        }
        return when (dir) {
            Direction.Left -> setSkin(SkinConfig.Player.skin_hit_left_path, Direction.Left)
            Direction.Right -> setSkin(SkinConfig.Player.skin_hit_right_path, Direction.Right)
            Direction.Up -> setSkin(
                    if (lastLR === Direction.Right) SkinConfig.Player.skin_up_right_path
                    else SkinConfig.Player.skin_up_left_path,
                    Direction.Up,
                    Direction.Non)
            Direction.Down -> setSkin(SkinConfig.Player.skin_down_left_path,
                                      Direction.Down,
                                      Direction.Non)
            Direction.Non -> setSkinToDefault()
        }
    }

    fun setSkinToDefault() {
        if (lastLR === Direction.Right) setCurrentImage(SkinConfig.Player.skin_standard_right_path)
        else setCurrentImage(SkinConfig.Player.skin_standard_left_path)
        direction = Direction.Non
    }

    init {
        setSkinToDefault()
    }
}
